File: backend/oneride/src/main/java/one/oneride/service/NotificationService.java
package one.oneride.service;

import java.util.List;

import one.oneride.dto.NotificationResponse;
import one.oneride.entity.User;
import one.oneride.enums.NotificationType;

public interface NotificationService {

    // =========================================================
    // CREATE GENERAL NOTIFICATION
    // =========================================================

    void createNotification(
            User user,
            String title,
            String message
    );

    // =========================================================
    // CREATE NOTIFICATION WITH TYPE
    // =========================================================

    void createNotification(
            User user,
            String title,
            String message,
            NotificationType type
    );

    // =========================================================
    // GET ALL NOTIFICATIONS
    // =========================================================

    List<NotificationResponse> getMyNotifications(
            String phoneNumber
    );

    // =========================================================
    // GET UNREAD NOTIFICATIONS
    // =========================================================

    List<NotificationResponse> getUnreadNotifications(
            String phoneNumber
    );

    // =========================================================
    // GET UNREAD COUNT
    // =========================================================

    long getUnreadCount(
            String phoneNumber
    );

    // =========================================================
    // MARK ONE AS READ
    // =========================================================

    void markAsRead(
            Long notificationId,
            String phoneNumber
    );

    // =========================================================
    // MARK ALL AS READ
    // =========================================================

    void markAllAsRead(
            String phoneNumber
    );
}

File: frontend/src/services/notificationService.js
import api from "../api/axios";

/*
 * =========================
 * GET ALL NOTIFICATIONS
 * =========================
 */

export const getNotifications = async () => {
  const response = await api.get("/notifications");

  return response.data;
};


/*
 * =========================
 * GET UNREAD NOTIFICATIONS
 * =========================
 */

export const getUnreadNotifications = async () => {
  const response = await api.get(
    "/notifications/unread"
  );

  return response.data;
};


/*
 * =========================
 * GET UNREAD COUNT
 * =========================
 */

export const getUnreadCount = async () => {
  const response = await api.get(
    "/notifications/unread-count"
  );

  return response.data;
};


/*
 * =========================
 * MARK ONE NOTIFICATION AS READ
 * =========================
 */

export const markNotificationAsRead = async (
  notificationId
) => {
  const response = await api.put(
    `/notifications/${notificationId}/read`
  );

  return response.data;
};


/*
 * =========================
 * MARK ALL NOTIFICATIONS AS READ
 * =========================
 */

export const markAllNotificationsAsRead = async () => {
  const response = await api.put(
    "/notifications/read-all"
  );

  return response.data;
};
